python/taichi/lang/matrix.py

```diff
@@ -539,31 +505,32 @@ def to_torch(self, as_vector=False, device=None):
         ti.sync()
         return ret
 
-    def from_numpy(self, ndarray):
-        if len(ndarray.shape) == self.loop_range().dim() + 1:
-            as_vector = True
-            assert self.m == 1, "This matrix is not a vector"
-        else:
-            as_vector = False
-            assert len(ndarray.shape) == self.loop_range().dim() + 2
+    def from_numpy(self, ndarray, keep_dims=False):
```

Comment:
We can probably infer `keep_dims` based on the dimensionality of `self` and `ndarray`. Maybe we should remove the `keep_dims` argument here?